pkg/apis/serving/v1alpha1/llm_inference_service_types.go

```diff
@@ -0,0 +1,304 @@
+
+	// WorkloadSpec configurations for the primary inference deployment.
+	// In a standard setup, this defines the main model server deployment.
+	// In a disaggregated setup (when 'prefill' is specified), this configures the 'decode' workload.
+	// +optional
+	WorkloadSpec `json:",inline,omitempty"`
+
+	// Router configuration for how the service is exposed. This section dictates the creation and management
+	// of networking resources like Ingress or Gateway API objects (HTTPRoute, Gateway).
+	// +optional
+	Router *RouterSpec `json:"router,omitempty"`
+
+	// Prefill configuration for disaggregated serving.
+	// When this section is included, the controller creates a separate deployment for prompt processing (prefill)
+	// in addition to the main 'decode' deployment, inspired by the llm-d architecture.
+	// This allows for independent scaling and hardware allocation for prefill and decode steps.
+	// +optional
+	Prefill *WorkloadSpec `json:"prefill,omitempty"`
+
+	// BaseRefs allows inheriting and overriding configurations from one or more LLMInferenceServiceConfig instances.
+	// The controller merges these base configurations, with the current LLMInferenceService spec taking the highest precedence.
+	// When multiple baseRefs are provided, the last one in the list overrides previous ones.
+	// +optional
+	BaseRefs []corev1.LocalObjectReference `json:"baseRefs,omitempty"`
+}
+
+// WorkloadSpec defines the configuration for a deployment workload, such as replicas and pod specifications.
+type WorkloadSpec struct {
+	// Number of replicas for the deployment.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Parallelism configurations for the runtime, such as tensor and pipeline parallelism.
+	// These values are used to configure the underlying inference runtime (e.g., vLLM).
+	// +optional
+	Parallelism *ParallelismSpec `json:"parallelism,omitempty"`
+
+	// Template for the main pod spec.
+	// In a multi-node deployment, this configures the "head" or "master" pod.
+	// In a disaggregated deployment, this configures the "decode" pod if it's the top-level template,
+	// or the "prefill" pod if it's within the Prefill block.
+	// +optional
+	Template *corev1.PodSpec `json:"template,omitempty"`
+
+	// Worker configuration for multi-node deployments.
+	// The presence of this field triggers the creation of a multi-node (distributed) setup.
+	// This spec defines the configuration for the worker pods, while the main 'Template' field defines the head pod.
+	// The controller is responsible for enabling discovery between head and worker pods.
+	// +optional
+	Worker *corev1.PodSpec `json:"worker,omitempty"`
+}
+
+// LLMModelSpec defines the model source and its characteristics.
+type LLMModelSpec struct {
+	// URI of the model, specifying its location, e.g., hf://meta-llama/Llama-4-Scout-17B-16E-Instruct
+	// The storage-initializer init container uses this URI to download the model.
+	URI apis.URL `json:"uri"`
```

Comment:
@yuzisun   Any reason to use storage uri as a string in the Inference service?

https://github.com/kserve/kserve/blob/master/pkg/apis/serving/v1beta1/predictor.go#L91

Reply:
I think it is confusing to `apis.URL` here as storage uri is not a HTTP url.